fairlearn/metrics/_metrics_engine.py

```diff
@@ -3,104 +3,124 @@
 
 import numpy as np
 
-from ._group_metric_result import GroupMetricResult
 from ._input_manipulations import _convert_to_ndarray_and_squeeze
+from sklearn.utils import Bunch
 
 _MESSAGE_SIZE_MISMATCH = "Array {0} is not the same size as {1}"
 
+# Parameters to metrics that should be split according to sensitive features
+_SPLITTABLE_METRIC_PARAMS = {"sample_weight"}
 
-def metric_by_group(metric_function,
-                    y_true, y_pred, group_membership,
-                    sample_weight=None,
-                    **kwargs):
+
+def group_summary(metric_function, y_true, y_pred,
+                  sensitive_features, **metric_params):
     r"""Apply a metric to each subgroup of a set of data.
 
-    :param metric_function: Function ``(y_true, y_pred, sample_weight=None, \*\*kwargs)``
+    :param metric_function: Function ``(y_true, y_pred, \*\*metric_params)``
 
     :param y_true: Array of ground-truth values
 
     :param y_pred: Array of predicted values
 
-    :param group_membership: Array Indicating the group to which each input value belongs
-
-    :param sample_weight: Optional weights to apply to each input value
+    :param sensitive_features: Array indicating the group to which each input value belongs
 
-    :param \*\*kwargs: Optional arguments to be passed to the `metric_function`
+    :param \*\*metric_params: Optional arguments to be passed to the `metric_function`
 
     :return: Object containing the result of applying ``metric_function`` to the entire dataset
-        and to each group identified in ``group_membership``.
-        If the ``metric_function`` returns a scalar, then additional fields are populated
-    :rtype: :class:`GroupMetricResult`
+        and to each group identified in ``sensitive_features``
+    :rtype: sklearn.utils.Bunch with the fields ``overall`` and ``by_group``
```

Comment:
Nit: I believe that there's a bit of ReST magic which can be inserted here to generate the link to `sklearn`. Will have to dig it out.

Reply:
a small issue here is that i don't think that `sklearn.utils.Bunch` is in sklearn docs

Reply:
Yes, I believe that @adrinjalali filed an issue with `sklearn` over that. Looking to the future, I believe that what you'll need is:
":py:class:`sklearn.utils.Bunch`"

Reply:
The bunch object is now documented in master. It'll be on the stable version of the website in the next release.

Reply:
done.